recipes/samarium/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2022 Jai Bellare
+# See <https://opensource.org/licenses/MIT/> or LICENSE.md
+# Project homepage: <https://github.com/strangeQuark1041/samarium>
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class SamariumConan(ConanFile):
+    name = "samarium"
+    version = "1.0.0"
+    description = "2-D physics simulation library"
+    homepage = "https://strangequark1041.github.io/samarium/"
+    url = "https://github.com/conan-io/conan-center-index/"
+    license = "MIT"
+    topics = ("c++20", "physics", "2d", "simulation")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    generators = "cmake_find_package"
+    requires = "fmt/8.1.1", "sfml/2.5.1"
+
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        self.options['sfml'].graphics = True
+        self.options['sfml'].window = True
+        self.options['sfml'].audio = False
+        self.options['sfml'].network = False
+
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
```

Comment:
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```
the source_folder should point to a place where exist a cmakefile.txt which call conan_basic_setup(), otherwise, conan won't be able to configure cmake